src/document/mod.rs

```diff
@@ -84,113 +116,225 @@ impl DocumentEncryptOpts {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let encrypted_data: Vec<u8> = vec![];
+    /// let decrypted_document = sdk.document_decrypt(&encrypted_data).await?;
+    /// let decrypted_data = decrypted_document.decrypted_data();
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_decrypt(&self, encrypted_document: &[u8]) -> Result<DocumentDecryptResult>;
+
+    /// Lists all of the IronCore encrypted documents that the calling user has been
+    /// granted access to.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # use ironoxide::document::DocumentListMeta;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// let document_data = sdk.document_list().await?;
+    /// let documents: Vec<DocumentListMeta> = document_data.result().to_vec();
+    /// # Ok(())
+    /// # }
     async fn document_list(&self) -> Result<DocumentListResult>;
 
-    /// Get the metadata for a specific document given its ID.
+    /// Returns the metadata for an IronCore encrypted document.
     ///
     /// # Arguments
     /// - `id` - Unique ID of the document to retrieve
     ///
-    /// # Returns
-    /// `DocumentMetadataResult` with details about the requested document.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// let document_data = sdk.document_get_metadata(&document_id).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn document_get_metadata(&self, id: &DocumentId) -> Result<DocumentMetadataResult>;
 
-    /// Attempt to parse the document ID out of an encrypted document.
+    /// Returns the document ID from the bytes of an IronCore encrypted document.
+    ///
+    /// This is the same ID returned by `DocumentEncryptResult.id()`.
     ///
     /// # Arguments
-    /// - `encrypted_document` - Encrypted document bytes
+    /// - `encrypted_document` - Bytes of the encrypted document
+    ///
+    /// # Errors
+    /// Fails if the provided encrypted bytes have no header.
     ///
-    /// # Returns
-    /// `Result<DocumentId>` Fails if provided encrypted document has no header, otherwise returns extracted ID.
+    /// # Examples
+    /// ```
+    /// # fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let bytes: Vec<u8> = vec![];
+    /// let document_id = sdk.document_get_id_from_bytes(&bytes)?;
+    /// # Ok(())
+    /// # }
+    /// ```
     fn document_get_id_from_bytes(&self, encrypted_document: &[u8]) -> Result<DocumentId>;
 
-    /// Encrypt the provided document bytes.
+    /// Updates the contents of an existing IronCore encrypted document.
```

Comment:
```suggestion
    /// Updates the contents of an existing encrypted document while preserving the documents metadata and user and group access control.
```

Reply:
Rust documentation guidelines seem to want a brief summary of the function followed by additional notes/information on the following lines. So even though it would flow fine as a single sentence, I think it makes it too long for a brief summary.

Reply:
Ok, 👍